packages/core/state/selection/reducer.ts

```diff
@@ -96,13 +98,24 @@ export default makeReducer<SelectionStateBranch>(
             ...state,
             fileGridColumnCount: action.payload,
         }),
-        [SET_FILE_FILTERS]: (state, action) => ({
-            ...state,
-            filters: action.payload,
+        [SET_FILE_FILTERS]: (state, action) => {
+            const recentAnnotations = Array.from(
```

Comment:
This could be simpler, also removes unnecessary castArray:
```
        [SET_FILE_FILTERS]: (state, action) => ({
            ...state,
            filters: action.payload,
            recentAnnotations: uniq([
                    ...action.payload.map((filter: any) => filter.annotationName),
                    ...state.recentAnnotations,
             ]),
            // Reset file selections when file filters change
            fileSelection: new FileSelection(),
       }),
```